ReactAndroid/src/main/java/com/facebook/react/uimanager/NativeViewHierarchyManager.java

```diff
@@ -383,7 +390,48 @@ public void manageChildren(
                       viewsToAdd,
                       tagsToDelete));
         }
-        dropView(viewToDestroy);
+
+        if (mLayoutAnimator.shouldAnimateLayout(viewToDestroy)) {
+          disableUserInteractions(viewToDestroy);
+          mLayoutAnimator.deleteView(viewToDestroy, new Callback() {
+            @Override
+            public void invoke(Object... args) {
+              ViewGroup parent = (ViewGroup)viewToDestroy.getParent();
+              if (parent != null) {
+                viewManager.removeView(parent, viewToDestroy);
+              }
+              dropView(viewToDestroy);
+            }
+          });
+        } else {
+          dropView(viewToDestroy);
+        }
+      }
+    }
+  }
+
+  /**
+   * Checks if a view's tag is contained in the tagsToDelete array.
+   */
+  private boolean isViewDeleted(View view, int[] tagsToDelete) {
```

Comment:
Used `Arrays.asList` instead, went for the array loop initially for perf issues but I verified and it is O(1) so it shouldn't be an issue especially since it is only ran when `shouldAnimateLayout` is true.